### DISCUSSION TITLE
Cannot Run/Debug libraries tests inside VS 2022/2019

### DISCUSSION THREAD

**Discussion by pedrobsaila** [dotnet/runtime#56247] [Q&A]
Hi (I don't know if discussions are the good way for this kind of requests) I'm having issues when trying to debug/run runtime libraries tests inside VS 2022/2019. Whenever I do it (like mentioned here https://github.com/dotnet/runtime/blob/main/docs/workflow/testing/visualstudio.md), the tests are skipped, and VS outputs the following message:

Microsoft.VisualStudio.TestPlatform.ObjectModel.TestPlatformException: Testhost process exited with error: Could not resolve CoreCLR path. For more details, enable tracing by setting COREHOST_TRACE environment variable to 1
. Please check the diagnostic logs for more information.
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.ThrowOnTestHostExited(Boolean testHostExited)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyOperationManager.SetupChannel(IEnumerable`1 sources, String runSettings)
   at Microsoft.VisualStudio.TestPlatform.CrossPlatEngine.Client.ProxyDiscoveryManager.DiscoverTests(DiscoveryCriteria discoveryCriteria, ITestDiscoveryEventsHandler2 eventHandler)

I joined VS detailed logs with  COREHOST_TRACE = 1

VS ouput window: [VS-ouput.log](https://github.com/dotnet/runtime/files/6870940/VS-ouput.log)

VS Logs: [logs.txt](https://github.com/dotnet/runtime/files/6870971/logs.txt)

Inside VS logs I have this strange error:

File name: 'System.Runtime.Extensions, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime.Extensions, Version=4.2.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.

It's strange because on nuget.org the latest version for System.Runtime.Extensions is 4.1.1.0

Do you have any ideas on how to fix that ?

**Comment by huoyaoyuan** [comment]
Did you build the runtime for enough components? I remember that at list `build libs+libs.tests` and `build libs.tests -f net472` is required.

**Reply by pedrobsaila** [reply]
Hi @huoyaoyuan thanks for taking time to answer me, I did just `build libs+libs.tests`. After trying also `build libs.tests -f net472` I still got the error.
